conan/tools/system/package_manager.py

```diff
@@ -147,7 +148,12 @@ def _install_substitutes(self, *packages_substitutes, update=False, check=True,
         raise ConanException("None of the installs for the package substitutes succeeded.")
 
     def _install(self, packages, update=False, check=True, **kwargs):
-        if check:
+        pkgs = self._conanfile.system_requires.setdefault(self._active_tool, [])
+        pkgs.extend(p for p in packages if p not in pkgs)
+        if self._mode == self.mode_collect:
+            return
+
+        if check or self._mode == self.mode_check:
```

Comment:
@czoido I am not sure if this was a bug or on-purpose, but ``apt`` was not running checks for mode=="check"?

Reply:
But it has a problem, ``apt-get`` is not doing even a check, but always returning the ``install()`` packages list, because the override has different default: `` def install(self, packages, update=False, check=False, recommends=False):``

Reply:
Finally did:
- ``mode=check`` always does a check, raise if packages not found. Can be done in ``graph info`` too
- ``mode=report`` do nothing except collect and report the system_requires. Do not check, do not raise.